thentos-core/src/Thentos/Types.hs

```diff
@@ -89,14 +89,18 @@ class ( Typeable dbParent, Typeable dbChild
     focus :: Lens' dbChild dbParent
     -- ^ Apply anything that is intended for @dbParent@ to @dbChild@.
 
-    asDBThentosError :: ThentosError dbParent -> ThentosError dbChild
+    thentosErrorFromParent :: ThentosError dbParent -> ThentosError dbChild
     -- ^ If a transaction or action associated with 'dbParent' throws an error, use
     -- this function to convert it to an error that can be thrown by
     -- transactions or actions associated with @dbChild@.
 
+    thentosErrorToParent :: ThentosError dbChild -> Maybe (ThentosError dbParent)
```

Comment:
This seems very strange. Basically we've managed to allow functions that shouldn't be handling certain kinds of errors to do so, by mapping those errors to `Nothing`. That is, in my opinion, a lot of type safety lost.